nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1498,6 +1497,7 @@ def build_train_valid_test_datasets(self):
             is_dataset_built_on_rank = lambda: True
 
             mock_dataset = True if self.cfg.data.get("data_impl", "mmap") == "mock" else False
+            add_extra_token = False if self.cfg.data.get("no_seqlen_plus_one_input_tokens", False) else True
```

Comment:
self.cfg.data.get("no_seqlen_plus_one_input_tokens", False) is the same no? as long as this attribute is a bool?

Reply:
It looks very obscure indeed. @dimapihtar what do you want to achieve here?

Reply:
@jkamalu @jbaczek 
I think `no_seqlen_plus_one_input_tokens` = True that we want to disable adding extra token as it says in config:
`# Set to True to disable fetching (sequence length + 1) input tokens, instead get (sequence length) input tokens and mask the last token`

Reply:
To be precise - we are not asking about the config field purpose, but rather the weird syntax here. 
Why do you use ternary expression here instead of simply using the config value?
Ternary expression does an implicit cast to bool. Similarly does the `not` operator. So if you want to flip the boolean value of a config field, use `not` instead of ternary expression.